ProductsService/ProductsService.Infrastructure/Repositories/QueryRepository.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.EntityFrameworkCore;
+using ProductsService.Domain.Abstractions.Database;
+using ProductsService.Domain.Abstractions.Specifications;
+using ProductsService.Domain.Entities.Abstractions;
+using ProductsService.Infrastructure.Data;
+using System.Linq.Expressions;
+
+namespace ProductsService.Infrastructure.Repositories;
+
+internal class QueryRepository<T> : IQueryRepository<T> where T : class, IEntity
+{
+    private readonly QueryDbContext _dbContext;
+    private readonly DbSet<T> _entities;
+
+    public QueryRepository(QueryDbContext dbContext)
+    {
+        _dbContext = dbContext;
+        _entities = _dbContext.Set<T>();
+    }
+
+    public async Task<T?> GetByIdAsync(int id, CancellationToken cancellationToken = default,
+        params Expression<Func<T, object>>[] includedProperties)
+    {
+        var query = _entities.AsQueryable();
+
+        foreach (var property in includedProperties)
+        {
+            query = query.Include(property);
+        }
```

Comment:
it's used almost in all methods, so what do you think about moving it to a separate method? maybe even extension method for IQueryable?